src/main/java/com/socialbox/service/impl/GroupServiceImpl.java

```diff
@@ -83,26 +84,28 @@ public Group createGroup(Group group) {
 
   @Override
   public List<GroupMovie> saveMovie(List<GroupMovie> groupMovies) {
+    if (groupMovies.isEmpty()) {
+      log.error("Empty list to be saved");
+      return new ArrayList<>();
+    }
+
+    Optional<Group> groupOptional = this.groupRepository.findById(groupMovies.get(0).getGroupId());
```

Comment:
We will not be storing different movies to different group, for now. This request will be mostly made to a single group.